Support for tagging and cross account access

excessivedemon opened the pull request:
* Added support for tagging (filtering volumes to backup based on their tags) via the --tag parameter

Examples of usage:

- to backup volumes with a tag of "Name" and value of "foo": `backup-monkey --region us-west-2 --tag Name:foo`

- to backup volumes with a tag of "Name" and value of "foo", and tag of "environment" and value of "production": `backup-monkey --region us-west-2 --tag Name:foo environment:production`

- to backup volumes with a tag of "Name" and value of "foo" or "bar": `backup-monkey --region us-west-2 --tag Name:"['foo','bar']"`

* Added support for --reverse-tags (Use the --tags parameter as a blacklist instead of a whitelist)

Example of usage:

- to snapshot all instances with a tag `Name` and value is NOT `foo`: `backup-monkey --region us-west-2 --tag Name:foo --reverse-tags`

* Added support for cross account snapshots. Added the following parameters:

`--cross-account-number CROSS_ACCOUNT_NUMBER`

Do a cross-account snapshot. Requires that you pass in the --cross-account-role parameter

`--cross-account-role CROSS_ACCOUNT_ROLE`

The role backup-monkey will assume when doing a cross-account snapshot

* Setup RPM builds to require python, python-setuptools, and RPM install to require python-boto. Setup post-install to install pip.

* Updated codes to run on Python 2.6 (Boto supports Python 2.6)

pas256 commented on README.rst:
```diff
@@ -36,8 +39,19 @@ Usage
+                            the --cross-account-role parameter
+      --cross-account-role CROSS_ACCOUNT_ROLE
+                            The role backup-monkey will assume when doing a cross-
+                            account snapshot
```
Can you please add an example here. It is not clear if you want the name of the role or the ARN.

pas256 commented on setup.py:
```diff
@@ -63,8 +68,13 @@
         'Intended Audience :: System Administrators',
         'License :: OSI Approved :: Apache Software License',
         'Natural Language :: English',
-        'Programming Language :: Python :: 2.7',
```
Keep both of these here, 2.7 and 2.6.
Thanks.

pas256 commented:
Thanks @excessivedemon. Let me know once those changes are taken care of and I'll merge.

Are you going to be at re:invent?

excessivedemon commented:
Thanks @pas256 I have now added examples for the --cross-account-number and --cross-account-role parameters, and updated the README.rst and setup.cfg files. There's also a small bug fix I added for the cross account functionality.

I'm currently in New Zealand but will be watching via live stream :smile:

pas256 commented:
Wicked. Thanks
